cronjob-scripts/cronjob_scripts/trigger_package_build.py

```diff
@@ -0,0 +1,196 @@
+    repo_url = get_repo_url()
+    version = get_current_version_if_unbuilt(
+        repo_url=repo_url, crate=crate, target=target, requested_version=requested_version
+    )
+    if not version:
+        return False
+
+    no_default_features = "false"
+    if crate == "gitoxide":
+        features = "max-pure"
+        no_default_features = "true"
+    elif crate == "sccache":
+        features = "vendored-openssl"
+    else:
+        features = ",".join(
+            feat
+            for feat in version["features"].keys()
+            if "vendored" in feat or "bundled" in feat
+        )
+    build_os = get_build_os(target)
+    branch = get_branch()
+    workflow_run_input = {
+        "crate": crate,
+        "version": version["vers"],
+        "target": target,
+        "features": features,
+        "no_default_features": no_default_features,
+        "build_os": build_os,
+        "branch": branch,
+    }
+    print(f"Attempting to build {crate} {version['vers']} for {target}")
+    subprocess.run(
+        ["gh", "workflow", "run", "build-package.yml", "--json", f"--ref={branch}"],
+        input=json.dumps(workflow_run_input).encode(),
+        check=True,
+    )
+    time.sleep(30)
+    return True
+
+
+@lru_cache
+def get_excludes(tracking_worktree_path: str, days: int, max_failures: int) -> set[str]:
+    """
+    if a crate has reached `max_failures` failures in the last `days` days then we exclude it
+    """
+    failures = Counter()
+
+    failure_filenames = list(Path(tracking_worktree_path).glob("*-*-*"))
+    failure_filenames.sort(reverse=True)
+
+    for failure_filename in failure_filenames[:days]:
+        with open(failure_filename, "r") as file:
+            failures.update(
+                Counter(stripped for line in file if (stripped := line.strip()))
+            )
+
+    return set(crate for crate, count in failures.items() if count >= max_failures)
+
+
+def get_repo_url() -> str:
+    repo_env_var = os.environ.get("GITHUB_REPOSITORY")
+    if repo_env_var:
+        return f"https://github.com/{repo_env_var}"
+    return subprocess.run(
+        ["gh", "repo", "view", "--json", "url", "--jq", ".url"],
+        capture_output=True,
+        text=True,
+    ).stdout.strip()
+
+
+def get_branch() -> str:
+    return subprocess.run(
+        ["git", "branch", "--show-current"],
+        capture_output=True,
+        text=True,
+    ).stdout.strip()
+
+
+def get_current_version_if_unbuilt(
+    repo_url: str,
+    crate: str,
+    target: str,
+    requested_version: str | None = None,
+) -> CrateVersionDict | None:
+    
+    # FIXME(probably never): in theory we could skip a round-trip to crates.io if we already know the version,
+    # but the code ends up looking ugly so I decided not to bother.
+    version = get_latest_version(crate, requested_version)
+    if not version:
+        return None
+
+    url = f"{repo_url}/releases/download/{crate}-{version['vers']}/{crate}-{version['vers']}-{target}.tar.gz"
```

Comment:
I think we might want to provide github token, shall save us a lot headache from rate limiting

Reply:
I switched this to use `gh release view $tag --json=assets` and catch "release not found"